xarray/coding/variables.py

```diff
@@ -152,18 +151,28 @@ def encode(self, variable, name=None):
         fv = encoding.get("_FillValue")
         mv = encoding.get("missing_value")
 
-        if fv is not None and mv is not None and not equivalent(fv, mv):
+        if (
+            fv is not None
+            and mv is not None
+            and not duck_array_ops.allclose_or_equiv(fv, mv)
+        ):
             raise ValueError(
                 "Variable {!r} has multiple fill values {}. "
                 "Cannot encode data. ".format(name, [fv, mv])
             )
 
         if fv is not None:
+            encoding["_FillValue"] = np.asarray(encoding["_FillValue"]).astype(
```

Comment:
Maybe using a NumPy scalar would work here? e.g., `encoding["_FillValue"] = data.dtype.type(encoding["_FillValue"])`

Actually I think you'll need to modify the `pop_to` line below, which overrides `_FillValue` in encoding.

Reply:
Thank you for the hint, @shoyer! 

> Actually I think you'll need to modify the pop_to line below, which overrides _FillValue in encoding.

I took a look at `pop_to()` function, and it appears that it is meant to just move a key/value from one dict to another dict. If I understand this correctly, we need to do type casting prior the `pop_to()` call, right?